cfme/storage/manager.py

```diff
@@ -0,0 +1,233 @@
+
+
+class ManagerAllView(ManagerView):
+    """The all Storage Manager or Provider page"""
+    @property
+    def is_displayed(self):
+        return (
+            self.in_manager and
+            self.title.text in ('Storage Managers', self.context['object'].manager_type))
+
+    toolbar = View.nested(ManagerToolbar)
+    entities = View.nested(ManagerEntities)
+    paginator = PaginationPane()
+
+
+class ManagerDetailsView(ManagerView):
+    """The details page for Storage Manager or Provider"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+
+        return(
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(ManagerDetailsToolbar)
+    sidebar = View.nested(ManagerDetailsAccordion)
+    entities = View.nested(ManagerDetailsEntities)
+
+
+class ManagerTagsView(ManagerView):
+    """The tag page for Storage Manager or Provider"""
+    breadcrumb = BreadCrumb()
+    select_tag = BootstrapSelect('tag_cat')
+    select_value = BootstrapSelect('tag_add')
+    save = Button('Save')
+    reset = Button('Reset')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_manager and
+            self.breadcrumb.active_location == 'Tag Assignment')
+
+
+class ManagePoliciesView(ManagerView):
+    """The policies page for Storage Manager or Provider"""
+    breadcrumb = BreadCrumb()
+    policies = BootstrapTreeview("protectbox")
+    save_button = Button("Save")
+    reset_button = Button("Reset")
+    cancel_button = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_manager and
+            self.breadcrumb.active_location == "'Storage Manager' Policy Assignment")
+
+
+class BlockManagerCollection(BaseCollection):
+    """Collection object [block manager] for the :py:class:'cfme.storage.manager'"""
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+        self.nav = VersionPick({
+            Version.lowest(): ['Storage', 'Storage Providers'],
+            '5.8': ['Storage', 'Block Storage', 'Managers']})
+        self.manager_type = 'Block Storage Managers'
+
+    def instantiate(self, name, provider):
+        return BaseManager(self, name, provider)
+
+
+class ObjectManagerCollection(BaseCollection):
+    """Collection object [object manager] for the :py:class:'cfme.storage.manager'"""
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+        self.nav = VersionPick({
+            Version.lowest(): ['Storage', 'Storage Providers'],
+            '5.8': ['Storage', 'Object Storage', 'Managers']})
+        self.manager_type = 'Object Storage Managers'
+
+    def instantiate(self, name, provider):
+        return BaseManager(self, name, provider)
+
+
+class BaseManager(BaseEntity):
```

Comment:
REQUIRED: can you please use some self explaining name here ? for example StorageManager, etc.

If you run into BaseManager class somewhere in another file, you won't be able to get what is that.
So, class names or public properties should have self explaining names.